src/nixenv.rs

```diff
@@ -0,0 +1,545 @@
+                .set_nix_options(self.nss_ca_cert_path)?
+                .args([
+                    "build",
+                    "--expr",
+                    r#"
+                    derivation {
+                        name = "user-environment";
+                        system = "builtin";
+                        builder = "builtin:buildenv";
+                        derivations = [];
+                        manifest = builtins.toFile "env-manifest.nix" "[]";
+                    }
+                "#,
+                    "--out-link",
+                ])
+                .arg(&temporary_profile)
+                .output()
+                .await
+                .map_err(|e| NixEnvError::StartNixCommand("nix build-ing an empty profile", e))?;
+
+            if !output.status.success() {
+                return Err(NixEnvError::NixCommand(
+                    "nix build-ing an empty profile",
+                    output,
+                ));
+            }
+        }
+
+        if let Ok(canon_profile) = self.profile.canonicalize() {
+            tracing::info!("Duplicating the existing profile into the scratch profile");
+
+            let output = tokio::process::Command::new(self.nix_store_path.join("bin/nix-env"))
+                .process_group(0)
+                .arg("--profile")
+                .arg(&temporary_profile)
+                .arg("--set")
+                .arg(canon_profile)
+                .output()
+                .await
+                .map_err(|e| NixEnvError::StartNixCommand("nix build-ing an empty profile", e))?;
+
+            if !output.status.success() {
+                return Err(NixEnvError::NixCommand(
+                    "nix build-ing an empty profile",
+                    output,
+                ));
+            }
+        }
+
+        // Query packages that are already installed in the profile.
+        // Constructs a map of (store path in the profile) -> (hash set of paths that are inside that store path)
+        let mut installed_paths: HashMap<PathBuf, HashSet<PathBuf>> = HashMap::new();
+        {
+            let output = tokio::process::Command::new(self.nix_store_path.join("bin/nix-env"))
+                .process_group(0)
+                .arg("--profile")
+                .arg(&temporary_profile)
+                .args(["--query", "--installed", "--out-path", "--json"])
+                .stdin(std::process::Stdio::null())
+                .output()
+                .await
+                .map_err(|e| {
+                    NixEnvError::StartNixCommand("nix-env --query'ing installed packages", e)
+                })?;
+
+            if !output.status.success() {
+                return Err(NixEnvError::NixCommand(
+                    "nix-env --query'ing installed packages",
+                    output,
+                ));
+            }
+
+            let installed_pkgs: HashMap<String, PackageInfo> =
+                serde_json::from_slice(&output.stdout)?;
```

Comment:
better error please

Reply:
like what?

Reply:
like "failed to get json from profiles" or something, idk, just something that says "we were trying to get a map of packages in your profile but couldn't because the json failed to deserialize  for some reason" instead of  the standard serde_json error of "expected map, got string" or something